### PR TITLE
add Controls.DomainServices services to OS Ria solution

### PR DESCRIPTION
**Pull request by ahmed-salem-me**
I've also changed versions in nuget.spec so that it's not created in different root folder under nuget packages folder
instead we can have, 
C:\Users\user\.nuget\packages\opensilver.openriaservices.client
                                                                                                      \5.00
                                                                                                      \4.6

### CODE REVIEW

File: OpenRiaServices.NuGet/OpenSilver/OpenRiaServices.OpenSilver.Client.Core.nuspec [hunk 1]
```diff
@@ -1,7 +1,7 @@
 <?xml version="1.0" encoding="utf-8"?>
 <package xmlns="http://schemas.microsoft.com/packaging/2011/10/nuspec.xsd">
   <metadata>
-    <id>OpenSilver.OpenRiaServices.Client.Core.4.6</id>
+    <id>OpenSilver.OpenRiaServices.Client.Core</id>
```
Comment (barjonp) [inline OpenRiaServices.NuGet/OpenSilver/OpenRiaServices.OpenSilver.Client.Core.nuspec]: We need to keep 4.6 at the end as we are maintaining 2 OpenRia packages. One based on version 5 of OpenRia (the id of the NuGet package is OpenSilver.OpenRiaServices.Client.Core) and one based on the version 4.6 for Silverlight compatibility (NuGet package is OpenSilver.OpenRiaServices.Client.Core.4.6)
```suggestion
    <id>OpenSilver.OpenRiaServices.Client.Core.4.6</id>
```

File: OpenRiaServices.NuGet/OpenSilver/OpenRiaServices.OpenSilver.Client.nuspec [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <package xmlns="http://schemas.microsoft.com/packaging/2011/10/nuspec.xsd">
   <metadata>
-    <id>OpenSilver.OpenRiaServices.Client.4.6</id>
+    <id>OpenSilver.OpenRiaServices.Client</id>
```
Comment (barjonp) [inline OpenRiaServices.NuGet/OpenSilver/OpenRiaServices.OpenSilver.Client.nuspec]:
```suggestion
    <id>OpenSilver.OpenRiaServices.Client.4.6</id>
```

File: OpenRiaServices.NuGet/OpenSilver/OpenRiaServices.OpenSilver.Client.nuspec [hunk 3]
```diff
@@ -17,7 +17,7 @@
     <tags>WCF RIA Services RIAServices OpenRiaServices OpenSilver Silverlight</tags>
     <dependencies>
       <group>
-        <dependency id="OpenSilver.OpenRiaServices.Client.Core.4.6" version="$PackageVersion$"/>
+        <dependency id="OpenSilver.OpenRiaServices.Client.Core" version="$PackageVersion$"/>
```
Comment (barjonp) [inline OpenRiaServices.NuGet/OpenSilver/OpenRiaServices.OpenSilver.Client.nuspec]:
```suggestion
        <dependency id="OpenSilver.OpenRiaServices.Client.Core.4.6" version="$PackageVersion$"/>
```

File: OpenRiaServices.NuGet/OpenSilver/OpenSilver.OpenRiaServices.DomainDataSource.nuspec [hunk 4]
```diff
@@ -0,0 +1,28 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2011/10/nuspec.xsd">
+  <metadata>
+    <id>OpenSilver.OpenRiaServices.DomainDataSource</id>
```
Comment (barjonp) [inline OpenRiaServices.NuGet/OpenSilver/OpenSilver.OpenRiaServices.DomainDataSource.nuspec]:
```suggestion
    <id>OpenSilver.OpenRiaServices.DomainDataSource.4.6</id>
```

File: OpenRiaServices.NuGet/OpenSilver/OpenSilver.OpenRiaServices.DomainDataSource.nuspec [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    <requireLicenseAcceptance>false</requireLicenseAcceptance>
+    <description>OpenSilver.OpenRiaServices.DomainDataSource provides the DomainDataSource control for OpenSilver, providing a declarative model for binding UI controls to a DomainContext.</description>
+    <summary>Open RIA Services - DomainDataSource control - For use in OpenSilver projects</summary>
+    <!-- <releaseNotes>For release notes see https://github.com/OpenRIAServices/OpenRiaServices/releases</releaseNotes> -->
+    <copyright>Copyright (c) 2021, Userware. All rights reserved.</copyright>
+    <language>en-US</language>
+    <tags>WCF RIA Services RIAServices Silverlight DomainDataSource OpenRIAServices OpenSilver</tags>
+    <dependencies>
+      <group targetFramework="netstandard2.0">
+        <dependency id="OpenSilver.OpenRiaServices.Client.Core"  version="$PackageVersion$"/>
```
Comment (barjonp) [inline OpenRiaServices.NuGet/OpenSilver/OpenSilver.OpenRiaServices.DomainDataSource.nuspec]:
```suggestion
        <dependency id="OpenSilver.OpenRiaServices.Client.Core.4.6"  version="$PackageVersion$"/>
```

**Review by barjonp — CHANGES_REQUESTED** [review]
Can you delete the `BinaryTypeUtility.cs`, `ExceptionHandlingUtility.cs` and `TypeUtility.cs` files. We can just add the file as references to the csproj instead of duplicating them. (you can copy from line 29 to line 39 of this file: https://github.com/OpenSilver/OpenRiaServices/blob/support/4.6/OpenRiaServices.Controls.DomainServices/Framework/OpenRiaServices.Controls.DomainServices.csproj#L29).

**Comment by ahmed-salem-me** [comment]
@barjonp I've made the required changes

**Pull request merged**